ProcessLib/RichardsFlow/RichardsFlowFEM.h

```diff
@@ -0,0 +1,198 @@
+
+        Eigen::MatrixXd mass_mat_coeff = Eigen::MatrixXd::Zero(1, 1);
+        Eigen::MatrixXd K_mat_coeff = Eigen::MatrixXd::Zero(1, 1);
+
+        MathLib::PiecewiseLinearInterpolation const& interP_Pc =
+            *_process_data.curves.at("curveA");
+        MathLib::PiecewiseLinearInterpolation const& interP_Kr =
+            *_process_data.curves.at("curveB");
+
+        auto local_M = MathLib::createZeroedMatrix<NodalMatrixType>(
+            local_M_data, local_matrix_size, local_matrix_size);
+        auto local_K = MathLib::createZeroedMatrix<NodalMatrixType>(
+            local_K_data, local_matrix_size, local_matrix_size);
+        auto local_b = MathLib::createZeroedVector<NodalVectorType>(
+            local_b_data, local_matrix_size);
+
+        unsigned const n_integration_points =
+            _integration_method.getNumberOfPoints();
+        SpatialPosition pos;
+        pos.setElementID(_element.getID());
+        double P_int_pt = 0.0;
+        for (unsigned ip = 0; ip < n_integration_points; ip++)
+        {
+            pos.setIntegrationPoint(ip);
+            auto const& sm = _shape_matrices[ip];
+            auto const& wp = _integration_method.getWeightedPoint(ip);
+
+            NumLib::shapeFunctionInterpolate(local_x, sm.N, P_int_pt);
+
+            auto const K = _process_data.intrinsic_permeability(t, pos)[0];
+            auto const poro = _process_data.porosity(t, pos)[0];
+            auto const mu = _process_data.viscosity(t, pos)[0];
+            auto const storage = _process_data.storage(t, pos)[0];
+            auto const rho_w = _process_data.water_density(t, pos)[0];
+
+            double Pc = -P_int_pt;
+            double Sw(1.0);
+            double dSwdPc(0.0);
+
+            Sw = interP_Pc.getValue(Pc);
+            dSwdPc = interP_Pc.getDerivative(Pc);
+            if (Pc > interP_Pc.getSupportMax())
+                dSwdPc = interP_Pc.getDerivative(interP_Pc.getSupportMax());
+            else if (Pc < interP_Pc.getSupportMin())
+                dSwdPc = interP_Pc.getDerivative(interP_Pc.getSupportMin());
+
+            _saturation[ip] = Sw;
+
+            double k_rel = interP_Kr.getValue(Sw);
+            double drhow_dp(0.0);
+
+            mass_mat_coeff(0, 0) =
+                storage * Sw + poro * Sw * drhow_dp - poro * dSwdPc;
+            K_mat_coeff(0, 0) = K * k_rel / mu;
+
+            local_K.noalias() += sm.dNdx.transpose() * K_mat_coeff(0, 0) *
+                                 sm.dNdx * sm.detJ * sm.integralMeasure *
+                                 wp.getWeight();
+
+            local_M.noalias() += sm.N.transpose() * mass_mat_coeff(0, 0) *
+                                 sm.N * sm.detJ * wp.getWeight();
+            if (_process_data.has_gravity)
+            {
+                typename ShapeMatricesType::GlobalDimVectorType vec_g;
+                vec_g.resize(_dim);
+                vec_g(_dim - 1) = -9.81;
```

Comment:
This quantity should be configurable from the input file.
See usage in [HMFem](https://github.com/endJunction/ogs/blob/HM/ProcessLib/HydroMechanics/HydroMechanicsFEM.h#L283) code and creation in [CreateHM](https://github.com/endJunction/ogs/blob/HM/ProcessLib/HydroMechanics/CreateHydroMechanicsProcess.cpp#L166) and in [prj file](https://github.com/ufz/ogs-data/blob/HM/HydroMechanics/Linear/square_1e1.prj#L35).